zql/docs/search-syntax/README.md

```diff
@@ -247,9 +247,24 @@ conn  1521912990.158539 ChhAfsfyuz4n2hFMe 10.239.34.35 56602     10.47.6.51 873
 
 ### Role of Data Types
 
-When working with named fields, the data type of the field comes becomes significant in two ways.
+When working with named fields, the data type of the field becomes significant in two ways.
 
-1. To match successfully, the value entered must be comparable to the data type of the named field. For instance, since `id.resp_h` is typically an `addr`-type field, an attempted field/value match `id.resp_h=10.150.0.999` will return an error, since this is not valid IP address syntax.
+1. To match successfully, the value entered must be comparable to the data type of the named field. For instance, the `host` field of the `http` events in our sample data are of `string` type, since it logs an HTTP header that is often a hostname or an IP address.
+
+   ```zq-command
+   zq -t 'count() by host | sort count,host' http.log.gz
+   ```
+
+   #### Output:
+   ```zq-output head:4
+   #0:record[host:bstring,count:uint64]
+   0:[0988253c66242502070643933dd49e88.clo.footprintdns.com;1;]
+   0:[10.47.21.1;1;]
+   0:[10.47.21.80/..;1;]
+   ...
+   ```
+
+   An attempted field/value match `host=10.47.21.1` would not match the event counted in the middle row of this table, since ZQL recognizes the bare value `10.47.21.1` as an IP address before comparing it to all the fields named `host` that it sees in the input stream. However, `host="10.47.21.1"` would match, since the quotes cause ZQL to treat the value as a string.
```

Comment:
@henridf: I made this change after seeing your PR review comment. Please give it another look.

Reply:
Reading this I'm thinking of `literal`  in place of `bare value`. But your call, I'm guessing you might find that it's too progamming language-y for the goal of this doc.

Reply:
Yes, I'm inclined to leave it as-is. I'm realizing that at some point we'll surely write a "Zesty Guide For Developers" where these same concepts are explained in programmer-friendly terms that use different terminology and use "foo/bar" examples rather than detailed Zeek ones that would make more sense to the security users. For now, I've got buy-in from the team that with our limited docs cycles we'll continue to make things as accessible as we can to security users who may not have a programming background. Thanks for helping me think it through.